src/Lean/Server/InfoUtils.lean

```diff
@@ -325,13 +325,6 @@ where
       return (some f!"```lean\n{fi.fieldName} : {tpFmt}\n```", none)
     | _ => return (none, none)
 
-  isAtomicFormat : Format → Bool
-    | Std.Format.text _    => true
-    | Std.Format.group f _ => isAtomicFormat f
-    | Std.Format.nest _ f  => isAtomicFormat f
-    | Std.Format.tag _ f   => isAtomicFormat f
-    | _                    => false
```

Comment:
I don't think that we want to delete this in general. It was [originally](https://github.com/leanprover/lean4/pull/522/commits/407f08c1cdab58e3dcd245d491d52d4987fda62a) introduced to avoid cluttering the hover panel with information already present below the cursor in non-`_` hover cases (e.g. the test cases in `4078.lean`) and I would like to retain that.

Reply:
Do you want this to still be used in this pr, or just kept around (unused) for future use?
My concern is that this heuristic can result in more-important expressions being hidden from the user - I first noticed this issue when looking at an `_` that was inferred to `(p - 1) / q`, which was harder for me to deduce by hand than if the term was something simpler like `p`.

Reply:
Ideally, the inferred expression would not be hidden away when hovering over `_`, but is hidden away in all other cases to avoid cluttering the hover panel with redundant information.